src/OrchardCore.Modules/OrchardCore.Workflows/Indexes/WorkflowIndexProvider.cs

```diff
@@ -10,7 +10,7 @@ public class WorkflowIndex : MapIndex
         public int DocumentId { get; set; }
         public string WorkflowTypeId { get; set; }
         public string WorkflowId { get; set; }
-        public int WorkflowStatus { get; set; }
+        public WorkflowStatus WorkflowStatus { get; set; }
```

Comment:
I already tried it after the previous update of yessql, where i think the enum type was already supported

But it was not working as the related column of the index table is still a string

Note: I tried to define here a string instead, and doing a toString() below, it was working, but i think it is more natural to hold  here an enum with an int. Maybe worth to tell it to @sebastienros as it was working before, by defining here an enum, and with column of string type.

Reply:
Hmm strange that it is a string in the index definition, that just feels wrong ;)

But also worked when I tried filtering by status. (with SQLite). I will check it with SQL Server.

I assume you were getting exceptions when you tried? No exceptions so far.

But also more type mapping has occured as well

Reply:
Need to leave i will check it again this night on sql server, if it works on your side

Reply:
Seems fine on sql. I leave you to check it tonight.

For these indexes, it should really be an int. Maybe worth updating on your document id pr? Perhaps also lengths for the typeid's as well. seems pointless to be 256, but I leave that idea to you to decide

Reply:
@deanmarcussen 

Okay cool it works on my side, maybe enum was not yet supported when i tried it

Reply:
@deanmarcussen 

> For these indexes, it should really be an int. Maybe worth updating on your document id pr?

Yes but e.g. with sqlite we can't alter a column in a migration step

> Perhaps also lengths for the typeid's as well. seems pointless to be 256, but I leave that idea to you to decide

Yes, i thought about this, but we can't alter a column length with sqlite in a migration step, and based on my new findings it seems to be easier that all new indexes can also be created for existing databases with the old lengths, and knowing that the number of rows in index pages only depends on the actual columns values sizes.

But yes, we could also change the columns sizes at least for new installations, but it open the possibility to define in the future a new index that can be created on a new installation, but would fail on a migrtion step.

But not against the idea ;)